examples/src/picking.ts

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import * as THREE from 'three';
+import CameraControls from 'camera-controls';
+import * as reveal from '@cognite/reveal';
+import * as reveal_threejs from '@cognite/reveal/threejs';
+
+CameraControls.install({ THREE });
+
+async function main() {
+  const modelUrl = new URL(location.href).searchParams.get('model') || '/primitives';
+
+  const pickedNodes: number[] = [];
+  const shading = reveal_threejs.createDefaultShading({
+    color(treeIndex: number) {
+      if (pickedNodes.indexOf(treeIndex) !== -1) {
+        return [255, 255, 0, 255];
+      }
+      return undefined;
+    }
+  });
+
+  const scene = new THREE.Scene();
+  const cadModel = await reveal.createLocalCadModel(modelUrl);
+  const cadNode = new reveal_threejs.CadNode(cadModel, { shading });
+
+  scene.add(cadNode);
+
+  const renderer = new THREE.WebGLRenderer();
+  renderer.setClearColor('#444');
+  renderer.setSize(window.innerWidth, window.innerHeight);
+  document.body.appendChild(renderer.domElement);
+
+  const { position, target, near, far } = cadNode.suggestCameraConfig();
+  const camera = new THREE.PerspectiveCamera(75, window.innerWidth / window.innerHeight, near, far);
+  const controls = new CameraControls(camera, renderer.domElement);
+  controls.setLookAt(position.x, position.y, position.z, target.x, target.y, target.z);
+  controls.update(0.0);
+  camera.updateMatrixWorld();
+  let pickingNeedsUpdate = false;
+  const clock = new THREE.Clock();
+  const render = async () => {
+    const delta = clock.getDelta();
+    const controlsNeedUpdate = controls.update(delta);
+    const sectorsNeedUpdate = await cadNode.update(camera);
+
+    if (controlsNeedUpdate || sectorsNeedUpdate || pickingNeedsUpdate) {
+      renderer.render(scene, camera);
+      pickingNeedsUpdate = false;
+    }
+
+    requestAnimationFrame(render);
+  };
+  render();
+
+  const pick = (event: MouseEvent) => {
+    const treeIndex = reveal_threejs.pickTreeIndex({ renderer, cadNode, scene, camera, event });
```

Comment:
Can this return position as well? And is an alternative approach to call it raypick that works either on a ray or a canvas XY position?

Reply:
I am looking returning position now, but will do it in a later PR. After looking into returning positions, having XY or rays as input, and supporting picking together with other things, I am realizing that we might need to do picking more like it is usually done in three.js (with more manual/advanced work). The simple "wrapper" version might have to be a lot more limited (not allowed to mix with other Three.js objects). But again, I think this should be done in a separate PR.